### ISSUE TITLE
How to sort a data-type="html" column?

### ISSUE THREAD

**Issue by caschbre** [fooplugins/FooTable#550]
I have a table that shows five-star rankings for different items.  The five star column header has a data-type="html" because the five-star widget is an ajax widget.  On each five-star cell I've added a data-sort-value="[integer]".  However this column is unable to sort on the five-star ranking.  If I change the header to data-type="number" then the sort works great, but the five-star ajax widget doesn't render properly.

Is there a way to set the sort type?  Or some other recommended way to sort a column like this?

**Comment by steveush** [comment]
Hi,

I've just tested this locally and I can't reproduce the issue you are having with sorting HTML columns, [here is a fiddle](https://jsfiddle.net/72m7epb7/) I created using the JS & CSS from this repo and it seems to work as expected.

PS: Please tell me which version you are running of the plugin.

Thanks

**Comment by caschbre** [comment]
Hi @steveush!  Thanks for taking a look.  Here's a fiddle of the issue I'm seeing.  I cloned your fiddle and dropped in my markup.  https://jsfiddle.net/0prmoav8/  

I'm running v3.1.1 standalone.

**Comment by steveush** [comment]
Hi @caschbre 

Did you update/save your cloned fiddle? The link you gave (https://jsfiddle.net/0prmoav8/) seems exactly the same as my original :)

Thanks

**Comment by caschbre** [comment]
@steveush Of course I did not save it... why would anyone expect to actually save changes?  :-D  Serves me right for doing this before I've had my coffee.

I inserted my markup again and clicked update this time. Here's the new link: https://jsfiddle.net/0prmoav8/1/

**Comment by steveush** [comment]
@caschbre Awesome, thanks for helping out, due to your updated fiddle I've found the issue and it is a simple falsy truthy problem when I retrieve the value for sorting. Basically to save on some lines I did the below:

``` javascript
var sortValue = $cell.data("sortValue") || cell.parser($cell);
```

Which is then seeing your value of 0 as false and so skips over to using the value of the parser which is basically the HTML for the cell and so your sort order gets all screwy. I'll make some updates today to change this to perform an explicit check on the data value and push them to the develop branch. I'll probably push a new version all together as there are a few fixes on that branch all ready which can go live.

Thanks
